core/templates/domain/topic/topic-rights-backend-api.service.spec.ts

```diff
@@ -16,206 +16,210 @@
-    TopicRightsBackendApiService = $injector.get(
-      'TopicRightsBackendApiService');
-    CsrfService = $injector.get('CsrfTokenService');
-    $rootScope = $injector.get('$rootScope');
-    $httpBackend = $injector.get('$httpBackend');
-
-    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
-      var deferred = $q.defer();
-      deferred.resolve('sample-csrf-token');
-      return deferred.promise;
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+
+import { TopicRightsBackendApiService } from
+  'domain/topic/topic-rights-backend-api.service';
+import { TopicRights } from 'domain/topic/topic-rights.model.ts';
+
+describe('Topic rights backend API service', () => {
+  let topicRightsBackendApiService: TopicRightsBackendApiService = null;
+  let httpTestingController: HttpTestingController;
+  let topicId = '0';
+  let topicName = '';
+  let topicRightsDict = {
+    can_edit_topic: false,
+    can_publish_topic: true,
+    published: true
+  };
+  let topicRightsObject = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [TopicRightsBackendApiService]
     });
-  }));
+
+    topicRightsBackendApiService = TestBed.get(TopicRightsBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+
+    topicRightsObject = TopicRights.createFromBackendDict(topicRightsDict);
+  });
 
   afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+    httpTestingController.verify();
   });
 
-  it('should fetch a topic rights', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should fetch a topic rights', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
 
-    $httpBackend.expect('GET', '/rightshandler/get_topic_rights/' + topicId)
-      .respond(200);
-    TopicRightsBackendApiService.fetchTopicRights(topicId).then(
+    topicRightsBackendApiService.fetchTopicRights(topicId).then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    var req = httpTestingController.expectOne(
+      '/rightshandler/get_topic_rights/' + topicId);
+    expect(req.request.method).toEqual('GET');
+    req.flush(200);
+
+    flushMicrotasks();
 
     expect(successHandler).toHaveBeenCalled();
     expect(failHandler).not.toHaveBeenCalled();
-  });
+  }));
 
-  it('should not fetch a topic rights', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should not fetch a topic rights', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
```

Comment:
const. Can you check for this sort of changes everywhere in the PR once?

Reply:
Done